File: app/src/main/java/org/tvheadend/tvhclient/domain/repository/data_source/TagAndChannelData.java
package org.tvheadend.tvhclient.domain.repository.data_source;

import androidx.lifecycle.LiveData;
import android.os.AsyncTask;
import androidx.annotation.NonNull;

import org.tvheadend.tvhclient.data.db.AppRoomDatabase;
import org.tvheadend.tvhclient.domain.entity.TagAndChannel;

import java.util.ArrayList;
import java.util.List;

public class TagAndChannelData implements DataSourceInterface<TagAndChannel> {

    private final AppRoomDatabase db;

    public TagAndChannelData(AppRoomDatabase database) {
        this.db = database;
    }

    @Override
    public void addItem(TagAndChannel item) {
        AsyncTask.execute(() -> db.getTagAndChannelDao().insert(item));
    }

    @Override
    public void updateItem(TagAndChannel item) {
        AsyncTask.execute(() -> db.getTagAndChannelDao().update(item));
    }

    @Override
    public void removeItem(TagAndChannel item) {
        AsyncTask.execute(() -> db.getTagAndChannelDao().delete(item));
    }

    public void addAndRemoveItems(List<TagAndChannel> newItems, List<TagAndChannel> oldItems) {
        AsyncTask.execute(() -> db.getTagAndChannelDao().insertAndDelete(
                new ArrayList<>(newItems),
                new ArrayList<>(oldItems)));
    }

    @Override
    public LiveData<Integer> getLiveDataItemCount() {
        return null;
    }

    @Override
    public LiveData<List<TagAndChannel>> getLiveDataItems() {
        return null;
    }

    @Override
    public LiveData<TagAndChannel> getLiveDataItemById(@NonNull Object id) {
        return null;
    }

    @Override
    public TagAndChannel getItemById(@NonNull Object id) {
        return null;
    }

    @Override
    @NonNull
    public List<TagAndChannel> getItems() {
        return new ArrayList<>();
    }

    public void removeItemByTagId(int id) {
        AsyncTask.execute(() -> db.getTagAndChannelDao().deleteByTagId(id));
    }
}

File: app/src/main/java/org/tvheadend/tvhclient/domain/repository/data_source/TagAndChannelData.kt
package org.tvheadend.tvhclient.domain.repository.data_source

import android.os.AsyncTask
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import org.tvheadend.tvhclient.data.db.AppRoomDatabase
import org.tvheadend.tvhclient.domain.entity.TagAndChannel
import java.util.*

class TagAndChannelData(private val db: AppRoomDatabase) : DataSourceInterface<TagAndChannel> {

    override fun addItem(item: TagAndChannel) {
        AsyncTask.execute { db.tagAndChannelDao.insert(item) }
    }

    override fun updateItem(item: TagAndChannel) {
        AsyncTask.execute { db.tagAndChannelDao.update(item) }
    }

    override fun removeItem(item: TagAndChannel) {
        AsyncTask.execute { db.tagAndChannelDao.delete(item) }
    }

    fun addAndRemoveItems(newItems: List<TagAndChannel>, oldItems: List<TagAndChannel>) {
        AsyncTask.execute {
            db.tagAndChannelDao.insertAndDelete(
                    ArrayList(newItems),
                    ArrayList(oldItems))
        }
    }

    override fun getLiveDataItemCount(): LiveData<Int> {
        return MutableLiveData()
    }

    override fun getLiveDataItems(): LiveData<List<TagAndChannel>> {
        return MutableLiveData()
    }

    override fun getLiveDataItemById(id: Any): LiveData<TagAndChannel> {
        return MutableLiveData()
    }

    override fun getItemById(id: Any): TagAndChannel? {
        return null
    }

    override fun getItems(): List<TagAndChannel> {
        return ArrayList()
    }

    fun removeItemByTagId(id: Int) {
        AsyncTask.execute { db.tagAndChannelDao.deleteByTagId(id) }
    }
}
